### PR TITLE
MVP tests for hass-util.html

### PR DESCRIPTION
**Pull request by emlove**
Want to get some feedback on this setup, before I continue rolling these out. It would have been nice to have some lighter unit-test framework to test the JS more isolated, since running an actual browser to test this is really overkill, but I couldn't find anything with my searching. Because of how Polymer is locked into HTML imports, we can't really make standalone JS imports (until Polymer 3).

### CODE REVIEW

**Comment by balloob** [comment]
Ugh, Polymer 3 can indeed not come fast enough.

One option could be that we already convert hassUtil to JS files and create a rollup bundle to add it to the window?

That way we can run independent tests against each function.

**Review by andrey-git — APPROVED** [review]

**Comment by andrey-git** [comment]
We could try move the code to a JS file and have html file import it. I think polymer-builder is now smart enough to inline it.

**Comment by armills** [comment]
I'll give it a try and see how it looks.

**Comment by armills** [comment]
I think this is the way we want to go. It lets us test our JS in isolation (lightning fast), and sets us up with a really clear path forward for Polymer 3.

File: src/util/hass-util.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <script src='../../bower_components/fecha/fecha.min.js'></script>
+<script src='../../build-temp/util.js'></script>
```
Comment (balloob) [inline src/util/hass-util.html]: Does this work with ES5 build?

File: package.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
     "dev-watch-es5": "npm run gulp watch_ru_all_es5 gen-service-worker-es5",
     "lint_js": "eslint src panels js --ext js,html",
     "lint_html": "ls -1 src/home-assistant.html panels/**/ha-panel-*.html | xargs polymer lint --input",
-    "test": "npm run lint_js && npm run lint_html"
+    "mocha": "BABEL_ENV=mocha node_modules/.bin/mocha --recursive --compilers js:babel-core/register test-mocha",
```
Comment (balloob) [inline package.json]: Let's create a file called `test-mocha/mocha.opts` for the command line flags, one per line:

```
--recursive
--compilers js:babel-core/register
--timeout 10000
```

File: src/util/hass-util.html (hunk 1 above)
Comment (balloob) [reply src/util/hass-util.html]: I guess it does as polymer-build will transpile for the ES5 build.

File: js/common/util/attribute_class_names.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export function attributeClassNames(stateObj, attributes) {
```
Comment (balloob) [inline js/common/util/attribute_class_names.js]: `export default`

File: package.json [hunk 4]
```diff
@@ -80,5 +82,12 @@
   },
   "devDependencies": {
     "web-component-tester": "^6.4.0"
+  },
+  "babel": {
+    "env": {
+      "mocha": {
+        "presets": ["es2015"]
```
Comment (balloob) [inline package.json]: Is this necessary? We're using node8 and that has native es2015. I think we only need to teach about import/export. You can achieve that with `reify`, see https://github.com/home-assistant/home-assistant-js-websocket/blob/master/test/mocha.opts#L1

File: src/util/hass-util.html (hunk 1 above)
Comment (armills) [reply src/util/hass-util.html]: I was under the impression that this gets transpiled by the polymer build process, but I don't have a complete understanding of all the moving pieces yet. Could definitely be wrong.

File: package.json (hunk 4 above)
Comment (armills) [reply package.json]: Nice! Good to know!

File: src/util/hass-util.html (hunk 1 above)
Comment (armills) [reply src/util/hass-util.html]: Ah, missed your reply. 👍

**Comment by andrey-git** [comment]
I think this introduces a race condition and possibly breaks the build script. Will check later :)

**Comment by andrey-git** [comment]
Removed comment. Wrong PR

**Pull request merged**